modules/spring-config/src/main/java/org/mule/runtime/config/spring/LazyMuleArtifactContext.java

```diff
@@ -97,12 +97,13 @@ private void applyLifecycle(List<String> createdComponentModels) {
     }
   }
 
+  /**
+   * During a lazy intialization of an artifact the components should not be created.
+   */
   @Override
   protected void createInitialApplicationComponents(DefaultListableBeanFactory beanFactory,
                                                     BeanDefinitionReader beanDefinitionReader) {
-    if (useNewParsingMechanism) {
-      createComponents(beanFactory, applicationModel, true);
-    } else {
+    if (!useNewParsingMechanism) {
```

Comment:
Just verify that test cases using an XML element that do not support the new mechanism works.